src/main/java/org/jitsi/videobridge/AbstractEndpoint.java

```diff
@@ -71,6 +87,70 @@
      */
     private boolean expired = false;
 
+    /**
+     * The set of IDs of the pinned endpoints of this {@code Endpoint}.
+     */
+    private Set<String> pinnedEndpoints = new HashSet<>();
+
+    /**
+     * The set of currently selected <tt>Endpoint</tt>s at this
+     * <tt>Endpoint</tt>.
+     */
+    private Set<String> selectedEndpoints = new HashSet<>();
+
+
+    /**
+     * Sets the list of pinned endpoints for this endpoint.
+     * @param newPinnedEndpoints the set of pinned endpoints.
+     */
+    public void pinnedEndpointsChanged(Set<String> newPinnedEndpoints)
+    {
+        // Check if that's different to what we think the pinned endpoints are.
+        Set<String> oldPinnedEndpoints = this.pinnedEndpoints;
+        if (!oldPinnedEndpoints.equals(newPinnedEndpoints))
+        {
+            this.pinnedEndpoints = newPinnedEndpoints;
+
+            if (logger.isDebugEnabled())
+            {
+                logger.debug("Pinned "
+                    + Arrays.toString(pinnedEndpoints.toArray()));
+            }
+
+            // bitrateController.setPinnedEndpointIds(pinnedEndpoints);
```

Comment:
should this be commented out? looks like it wasn't before...i'm assuming we don't have access to that here (in `AbstractEndpoint`), so maybe we'll still need some overridable method in the sub-classes so `Endpoint` can do this?

Reply:
Thanks for catching this, Brian. I just pushed a fix.